File: Projects/TP02NumberBaseball/app/src/main/java/com/hyun/tp02numberbaseball/MainActivity.java
package com.hyun.tp02numberbaseball;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import java.sql.Array;
import java.util.ArrayList;
import java.util.Random;


public class MainActivity extends AppCompatActivity {

    EditText et;
    Button btn;
    TextView tv;
    Random rnd;
    int com;
    int com2;
    int com3;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    et=findViewById(R.id.et);
    btn=findViewById(R.id.btn);
    tv=findViewById(R.id.tv);
    Random rnd= new Random();
    com=rnd.nextInt();
    com2=rnd.nextInt();
    com3=rnd.nextInt();

    btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int n1= Integer.parseInt(et.getText().toString());
                int n2= Integer.parseInt(et.getText().toString());
                int n3= Integer.parseInt(et.getText().toString());
                int strike=0; int ball=0;




            }
        });


















    }
}

File: Projects/PoketmonAPI/app/src/main/java/com/mrhi2024/poketmonapi/MainActivity.kt
package com.mrhi2024.poketmonapi

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import androidx.appcompat.app.AlertDialog
import com.google.gson.Gson
import org.json.JSONObject
import java.io.BufferedReader
import java.io.InputStreamReader
import java.net.URL
import kotlin.concurrent.thread

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.btn).setOnClickListener {
            thread {

                val url= URL("https://pokeapi.co/api/v2/pokemon/1")
                val inputStream= url.openStream()
                val inputStreamReader= InputStreamReader(inputStream)
                val reader= BufferedReader(inputStreamReader)

                val builder= StringBuilder()
                while (true){
                    val line= reader.readLine() ?: break
                    builder.append(line+"\n")
                    Log.d("aa", line)
                }

                //runOnUiThread { AlertDialog.Builder(this).setMessage(builder.toString()).create().show() }

                val s:String= builder.toString()

                val gson= Gson()
                val poketmonItem:PoketmonItem= gson.fromJson(s, PoketmonItem::class.java)

                runOnUiThread {  AlertDialog.Builder(this).setMessage("${poketmonItem.id} : ${poketmonItem.name}").create().show() }

//                val json:JSONObject= JSONObject(s)
//
//                val id:Int= json.getInt("id")
//                val name:String= json.getString("name")
//
//                runOnUiThread {  AlertDialog.Builder(this).setMessage("$id : $name").create().show() }

            }
        }
    }
}